Improve stubs for `flatten` and `flatten_fieldsets`

ngnpope opened the pull request:
# I have made things!

These functions are implemented in a generic way that doesn't generally care about the content. The existing return type of `Callable | str` makes these awkward to use.

I've changed the use of `Sequence` in the `_FieldGroups` alias to use `_ListOrTuple`. This is another case where `ModelAdmin.fields` and the `"fields"` item in fieldsets are presumed by Django to be a `list` or `tuple`, e.g. there are some system checks that ensure this.

See https://github.com/django/django/blob/afbb8c709d40e77b3f71c152d363c5ad95ceec2d/django/contrib/admin/utils.py#L102-L120

## Related issues

N/A

ngnpope committed:
Improve stubs for `flatten` and `flatten_fieldsets`

These functions are implemented in a generic way that doesn't
generally care about the content. The existing return type of
`Callable | str` makes these awkward to use.

I've changed the use of `Sequence` in the `_FieldGroups` alias to use
`_ListOrTuple`. This is another case where `ModelAdmin.fields` and the
`"fields"` item in fieldsets are presumed by Django to be a `list` or
`tuple`, e.g. there are some system checks that ensure this.

See https://github.com/django/django/blob/afbb8c709d40e77b3f71c152d363c5ad95ceec2d/django/contrib/admin/utils.py#L102-L120

sobolevn commented on django-stubs/contrib/admin/utils.pyi:
```diff
@@ -31,8 +31,8 @@ def prepare_lookup_value(
 def unquote(s: str) -> str: ...
-def flatten(fields: Any) -> list[Callable | str]: ...
-def flatten_fieldsets(fieldsets: Any) -> list[Callable | str]: ...
+def flatten(fields: _FieldGroups[_T]) -> list[_T]: ...
```
Can we please add a test case for this? It seems complex enough.

sobolevn reviewed (commented):
Thanks!

ngnpope committed:
Add new `_ListDisplayT` type alias

This renames `_DisplayT` to `_ListDisplayT` and creates a new
`_DisplayT` alias with the contents of the `_ListOrTuple`.

This will be used in the next commit.

sobolevn reviewed (commented):
Looks like pyright is not happy with something :(

sobolevn reviewed (approved):
Awesome! Thanks!
